Content.Server/Anomaly/Effects/ProjectileAnomalySystem.cs

```diff
@@ -0,0 +1,90 @@
+using Content.Server.Weapons.Ranged.Systems;
+using Content.Shared.Anomaly.Components;
+using Content.Shared.Anomaly.Effects.Components;
+using Content.Shared.Mobs.Components;
+using Content.Shared.Projectiles;
+using Robust.Server.GameObjects;
+using Robust.Shared.Map;
+using Robust.Shared.Random;
+
+namespace Content.Server.Anomaly.Effects;
+
+/// <summary>
+/// This handles <see cref="ProjectileAnomalyComponent"/> and the events from <seealso cref="AnomalySystem"/>
+/// </summary>
+public sealed class ProjectileAnomalySystem : EntitySystem
+{
+    [Dependency] private readonly TransformSystem _xform = default!;
+    [Dependency] private readonly EntityLookupSystem _lookup = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly IMapManager _mapManager = default!;
+    [Dependency] private readonly GunSystem _gunSystem = default!;
+
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<ProjectileAnomalyComponent, AnomalyPulseEvent>(OnPulse);
+        SubscribeLocalEvent<ProjectileAnomalyComponent, AnomalySupercriticalEvent>(OnSupercritical);
+    }
+
+    private void OnPulse(EntityUid uid, ProjectileAnomalyComponent component, ref AnomalyPulseEvent args)
+    {
+        ShootProjectilesAtEntites(uid, component, args.Severity, args.Stability);
+    }
+
+    private void OnSupercritical(EntityUid uid, ProjectileAnomalyComponent component, ref AnomalySupercriticalEvent args)
+    {
+        ShootProjectilesAtEntites(uid, component, 1.0f, 1.0f);
+    }
+
+    private void ShootProjectilesAtEntites(EntityUid uid, ProjectileAnomalyComponent component, float severity, float stability)
+    {
+        var xform = Transform(uid);
+        var projectilesShot = 0;
+        var range = Math.Abs(component.ProjectileRange * stability); // Apparently this shit can be a negative somehow?
+
+        foreach (var entity in _lookup.GetEntitiesInRange(uid, range, LookupFlags.Dynamic))
+        {
+            if (projectilesShot >= component.MaxProjectiles * severity)
+                return;
+
+            // Living entities are more likely to be shot at than non living
+            if (!HasComp<MobStateComponent>(entity) && !_random.Prob(component.TargetNonLivingChance))
+                continue;
```

Comment:
use a MobStateComponent query outside and just hascomp with that.

Reply:
I don't understand what you mean by this, I'm not just looking for entities with the MobStateComponent, querying over just entities with MobStateComponent would ignore other non living entities which is not the point of this code snippet